### PR TITLE
feat(grow): add residue beat graph model (Phase 8d)

### PR DESCRIPTION
**Pull request by pvliesdonk**
## Summary

Add the graph-level data model for residue beats — variant passages at convergence points where path-specific prose should differ. This is the foundation for Phase 8d (residue beat proposals) in PR4.

## Related Issues

Partial #853 (Design doc D2: graph model)
Part of Epic #858 (GROW stage redesign, Phase 2)

**PR Stack:** PR1 (#877, merged) → PR2 (#878) → **PR3 (this)** → PR4 → PR5

## Changes

- Add `ResidueVariant`, `ResidueBeatProposal`, `Phase8dOutput` Pydantic models to `models/grow.py`
- Add `ResidueCandidate` dataclass and `find_residue_candidates()` to scan convergence points for soft/flavor dilemmas
- Add `ResidueCreationResult` dataclass and `create_residue_passages()` to create variant passages gated by codewords
- Variant naming convention: `passage::{base}__via_{codeword_suffix}`
- Base passage preserved as fallback; variants get `is_residue=True`, `is_synthetic=True` metadata

## Not Included / Future PRs

- Phase 8d LLM prompt and phase function (PR4)
- Residue-aware passage collapse/absorption (PR4)
- Simplification of existing phases (PR5)

## Test Plan

- [x] 10 model tests: ResidueVariant validation, ResidueBeatProposal constraints, Phase8dOutput
- [x] 13 algorithm tests: find_residue_candidates (soft/flavor/hard policies, dedup, edge cases), create_residue_passages (creation, naming, validation, skipping)
- [x] 345 total tests pass (no regressions)
- [x] mypy and ruff clean

```
uv run pytest tests/unit/test_grow_models.py tests/unit/test_grow_algorithms.py -x -q -k residue
# 21 passed
```

## Checklist

- [x] Code follows project conventions
- [x] Tests added/updated for changes
- [x] No TODO stubs in committed code
- [x] Type hints added for new code

## Review Guide

1. `src/questfoundry/models/grow.py` — new Pydantic models (small, self-contained)
2. `src/questfoundry/graph/grow_algorithms.py` — `find_residue_candidates()` then `create_residue_passages()`
3. `tests/unit/test_grow_models.py` — model validation tests
4. `tests/unit/test_grow_algorithms.py` — algorithm tests with graph fixtures

### CODE REVIEW

File: src/questfoundry/graph/grow_algorithms.py [hunk 1]
```diff
@@ -1698,6 +1700,240 @@ def _beat_index(dc: DilemmaConvergence, seq: list[str] = branch_after_div) -> in
...
+def create_residue_passages(
+    graph: Graph,
+    proposals: list[dict[str, Any]],
+) -> ResidueCreationResult:
+    """Create variant passages from accepted residue proposals.
+
+    For each proposal:
+    1. Create variant passages (passage::{base}__via_{codeword_suffix})
+    2. Mark variants with is_residue=True and residue metadata
+    3. Wire choice edges routing through variants based on codewords
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/graph/grow_algorithms.py]: Fixed in 9909cfa. Updated docstring to remove step 3 (edge wiring) — that is Phase 9's responsibility, not this function's.

File: src/questfoundry/graph/grow_algorithms.py [hunk 2]
```diff
@@ -1698,6 +1700,240 @@ def _beat_index(dc: DilemmaConvergence, seq: list[str] = branch_after_div) -> in
...
+        if path_id:
+            path_codewords.setdefault(path_id, []).append(cw_id)
+
+    # Build dilemma → paths mapping
+    dilemma_paths: dict[str, list[str]] = {}
+    for path_id, pdata in path_nodes.items():
+        did = pdata.get("dilemma_id", "")
+        if did:
+            full_did = normalize_scoped_id(did, "dilemma")
+            dilemma_paths.setdefault(full_did, []).append(path_id)
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/graph/grow_algorithms.py]: Fixed in 9909cfa. Replaced with build_dilemma_paths(graph) and removed unused path_nodes variable.
